Changed to accept tooltips in components that use FormElement internally

kosuke-k opened the pull request:
Closes #451 

## What I did
- Add `tooltip` and `tooltipIcon` acceptance in components that use FormElement internally: 772046e7d9fde4e9d13484eb34af76769c602871
  - `Input/Textarea/Select/Picklist/Lookup/DateInput/Checkbox`
- Add `TooltipContent` to `CheckboxGroup` and `RadioGroup` to accept it: 2066d28440eb8b2a9a3cd6d95d8e6431aa408c92
- Add stories with tooltip

stomita reviewed (changes requested):
This PR makes VRT diffs of layout on Horizontal Form story.

https://reg-suit-react-lightning-design-system.s3.amazonaws.com/2066d28440eb8b2a9a3cd6d95d8e6431aa408c92/index.html?id=changed-Form/Horizontal%20Form.png

![CleanShot 2025-05-23 at 16 58 02@2x](https://github.com/user-attachments/assets/532f8e31-c8c0-4123-ad1c-1052146b8e4e)

kosuke-k commented:
> This PR makes VRT diffs of layout on Horizontal Form story.
> 
> https://reg-suit-react-lightning-design-system.s3.amazonaws.com/2066d28440eb8b2a9a3cd6d95d8e6431aa408c92/index.html?id=changed-Form/Horizontal%20Form.png

@stomita 
Fixed layout collapse of CheckboxGroup and RadioGroup on wide horizontal forms, caused by the addition of tooltips. Fixed in https://github.com/mashmatrix/react-lightning-design-system/pull/455/commits/994ce6a2f1a68590ab44628ffaf0e81762c7d4af.

The issue was due to wrapping the fields in a `<div>`, and was resolved by placing the tooltip inside the `<legend>` element.
